src/Adapter/Order/Refund/OrderRefundCalculator.php

```diff
@@ -97,7 +98,7 @@ public function computeOrderRefund(
             $refundedAmount = $voucherAmount = $chosenVoucherAmount;
         }
 
-        $shippingCostAmount = $shippingRefund ?: false;
+        $shippingCostAmount = (float) (string) $shippingRefund;
```

Comment:
I understand you need to cast twice. the first one will call __toString method and the second cast it to float.

Suggestion : Can't we add a method toFloat in Number class to do it in one place ?

Reply:
@sowbiba yes, that could be a solution. However it would means that we should update the Decimal library and IMO it's not worth it as we're planning to use Decimal everywhere we can instead of float in the coming releases.